src/wazuh_modules/wm_exec.c

```diff
@@ -616,58 +632,61 @@ void* reader(void *args) {
 // Add process group to pool
 
 void wm_append_sid(pid_t sid) {
-    int i;
 
-    w_mutex_lock(&wm_children_mutex);
+    pid_t * p_sid = NULL;
 
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (!wm_children[i]) {
-            wm_children[i] = sid;
-            break;
+    os_calloc(1, sizeof(pid_t), p_sid);
+    *p_sid = sid;
+
+    if (wm_children_list) {
+        if (OSList_AddData(wm_children_list, (void *)p_sid)) {
+            return;
         }
```

Comment:
I miss a mutex lock here.

```suggestion
        w_mutex_lock(&wm_children_mutex);
        void * retval = OSList_AddData(wm_children_list, (void *)p_sid);
        w_mutex_unlock(&wm_children_mutex);

        if (retval != NULL) {
            return;
        }
```